Ghidra/Processors/CP1600/data/languages/CP1600.slaspec

```diff
@@ -0,0 +1,708 @@
+	I = 1;
+}
+
+:GSWD reg0_1 is opcode2_9=0x000C & reg0_1 [ doublebyte=0; globalset(inst_next, doublebyte); ] {
+	local mask:2 = (zext(S) << 7) + (zext(Z) << 6) + (zext(O) << 5) + (zext(C) << 4);
+	reg0_1 = (mask << 16) + mask; 
+}
+
+:HLT is opcode0_9=0x0000 [ doublebyte=0; globalset(inst_next, doublebyte); ] {
+	goto inst_start;
+}
+
+:INCR reg0_2 is opcode3_9=0x0001 & reg0_2 [ doublebyte=0; globalset(inst_next, doublebyte); ] {
+	reg0_2 = reg0_2 + 1;
+	resultFlags(reg0_2);
+}
+
+:J jmpdest16 is opcode0_9=0x0004 ; jump_type=0 & reg24_25=R7 & jmpdest16 [ doublebyte=0; globalset(inst_next, doublebyte); ] {
+	goto jmpdest16;
+}
+
+:JD jmpdest16 is opcode0_9=0x0004 ; jump_type=2 & reg24_25=R7 & jmpdest16 [ doublebyte=0; globalset(inst_next, doublebyte); ] {
+	I = 0;
+	goto jmpdest16;
+}
+
+:JE jmpdest16 is opcode0_9=0x0004 ; jump_type=1 & reg24_25=R7 & jmpdest16 [ doublebyte=0; globalset(inst_next, doublebyte); ] {
+	I = 1;
+	goto jmpdest16;
+}
+
+:JSR reg24_25, jmpdest16 is opcode0_9=0x0004 ; jump_type=0 & reg24_25 & jmpdest16 [ doublebyte=0; globalset(inst_next, doublebyte); ] {
+	reg24_25 = inst_next;
+	call jmpdest16;
+}
+
+:JSRD reg24_25, jmpdest16 is opcode0_9=0x0004 ; jump_type=2 & reg24_25 & jmpdest16 [ doublebyte=0; globalset(inst_next, doublebyte); ] {
+	I = 0;
+	reg24_25 = inst_next;
+	call jmpdest16;
+}
+
+:JSRE reg24_25, jmpdest16 is opcode0_9=0x0004 ; jump_type=1 & reg24_25 & jmpdest16 [ doublebyte=0; globalset(inst_next, doublebyte); ] {
+	I = 1;
+	reg24_25 = inst_next;
+	call jmpdest16;
+}
+
+:MOVR reg3_5, reg0_2 is opcode6_9=0x0002 & reg0_2 & reg3_5 [ doublebyte=0; globalset(inst_next, doublebyte); ] {
+	reg0_2 = reg3_5;
+	resultFlags(reg0_2);
+}
+
+:MVI addr16, reg0_2 is opcode3_9=0x0050 & reg0_2 ; addr16 [ doublebyte=0; globalset(inst_next, doublebyte); ] {
+	reg0_2 = *:2 addr16;
+}
+
+:MVO reg0_2, addr16 is opcode3_9=0x0048 & reg0_2 ; addr16 [ doublebyte=0; globalset(inst_next, doublebyte); ] {
+	*addr16 = reg0_2;
+}
+
+:MVO@ reg3_5, reg0_2 is opcode6_9=0x0009 & reg0_2 & reg3_5 [ doublebyte=0; globalset(inst_next, doublebyte); ] { 
+	*:2 (reg3_5 * 2) = reg0_2;
+	if ((reg3_5 != R4) && (reg3_5 != R5) && (reg3_5 != R6)) goto <done>;
+	reg3_5 = reg3_5 + 1;
+	<done>
+}
+
+:MVOI reg0_2 is opcode3_9=0x004F & reg0_2 ; imm10 [ doublebyte=0; globalset(inst_next, doublebyte); ] {
+	local tmp:2 = inst_start + 2;
+	*tmp = reg0_2;
+}
+
+:NEGR reg0_2 is opcode3_9=0x0004 & reg0_2 [ doublebyte=0; globalset(inst_next, doublebyte); ] {
+	local tmp = reg0_2 ^ 0xFFFF;
+	local tmpC = carry(tmp, 1);
+	local tmpO = scarry(tmp, 1);
+	reg0_2 = -reg0_2;
+	C = tmpC;
+	O = tmpO;
+	resultFlags(reg0_2);
+}
+
+:NOP is opcode1_9=0x001A [ doublebyte=0; globalset(inst_next, doublebyte); ] {
+}
+
+:NOPP is opcode6_9=0x0008 & branch_external=0 & branch_condition=8 ; imm10 [ doublebyte=0; globalset(inst_next, doublebyte); ] {
+}
+
+:RSWD reg0_2 is opcode3_9=0x0007 & reg0_2 [ doublebyte=0; globalset(inst_next, doublebyte); ] {
```

Comment:
None of these can ever equal 1... != 0 maybe

Reply:
I'm not sure what you mean with that.  NOPP has a bitmask of `1000S01000 xxxxxxxxxx`, which breaks down as bits 6..9 being 0b1000, bit 5 being ignored (sign), bit 4 (branch_external) being 0 and bits 0..3 (branch_condition) being 0b1000, with an extra decle that should be skipped.  I can't see why this pattern wouldn't match...

Reply:
` (reg0_2 & 0b00001000) == 1;`. Will result in either 0==1 or 8==1 checks, neither can be true, so it looks like you meant for !=0

Reply:
Oh, I see.  The code being highlighted didn't really show the contents of RSWD.  That was a leftover from a copy and paste gone wrong.  I'll fix this right away.

Reply:
Yeah, I can see the confusion, sorry. Should have commented one line down

Reply:
No worries. This should be fixed as well - I'm still not happy about having to force a disassembly action on every opcode but I don't know if using `with` would have helped here for that.